packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -800,15 +811,15 @@ class WidgetTester extends WidgetController implements HitTestDispatcher, Ticker
         .map((HitTestEntry candidate) => candidate.target)
         .whereType<RenderObject>()
         .first;
-      final Element? innerTargetElement = collectAllElementsFrom(
-        binding.renderViewElement!,
-        skipOffstage: true,
-      ).cast<Element?>().lastWhere(
-        (Element? element) => element!.renderObject == innerTarget,
-        orElse: () => null,
+      final Element? innerTargetElement = _lastWhereOrNull(
+        collectAllElementsFrom(
+          binding.renderViewElement!,
+          skipOffstage: true,
+        ).cast<Element>(),
```

Comment:
Why this cast? `collectAllElementsFrom` already returns an `Iterable<Element>`, I think.